lsp-mode.el

```diff
@@ -5138,15 +5227,31 @@ Others: TRIGGER-CHARS"
 (defun lsp--resolve-completion (item)
   "Resolve completion ITEM."
   (cl-assert item nil "Completion item must not be nil")
-  (or (-first 'identity
-              (condition-case _err
-                  (lsp-foreach-workspace
-                   (when (lsp:completion-options-resolve-provider?
-                          (lsp--capability :completionProvider))
-                     (lsp-request "completionItem/resolve" item)))
-                (error)))
+  (or (ignore-errors
+        (when (lsp:completion-options-resolve-provider?
+               (lsp--capability :completionProvider))
+          (lsp-request "completionItem/resolve" item)))
       item))
 
+(defun lsp--resolve-completion-async (item callback &optional cleanup-fn)
```

Comment:
nit: is this really optional?

Reply:
Yes, the `cleanup-fn` can be optional, for example to resolve doc asynchronously (just assign properties to `item`)